WcaOnRails/app/webpacker/components/DelegateTab/RoleForm.jsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+
+import { Form } from 'semantic-ui-react';
+
+import delegateStatus from '../../lib/helpers/delegate-status';
+import I18n from '../../lib/i18n';
+import { delegateUpdateDelegateUrl } from '../../lib/requests/routes.js.erb';
+import { fetchWithAuthenticityToken } from '../../lib/requests/fetchWithAuthenticityToken';
+
+const notDelegateOption = 'None';
+
+function RoleForm({
+  values,
+  regionList,
+  disabled,
+}) {
+  const { userId, ...delegateDetails } = values;
+  const [savedValues, setSavedValues] = React.useState(values.status ? delegateDetails : {});
+  const [formValues, setFormValues] = React.useState(values.status ? delegateDetails : {});
+
+  const updateDelegate = async () => {
+    const url = formValues.status ? delegateUpdateDelegateUrl(
+      userId,
+      formValues.status,
+      formValues.location,
+      formValues.regionId,
+    ) : delegateUpdateDelegateUrl(userId);
+    await fetchWithAuthenticityToken(url);
+    setSavedValues(formValues);
+  };
+
+  return (
+    <Form onSubmit={() => updateDelegate()}>
+      <Form.Dropdown
+        label={I18n.t('activerecord.attributes.user.delegate_status')}
+        fluid
+        selection
+        value={formValues.status || notDelegateOption}
+        disabled={disabled}
+        options={[{
```

Comment:
This could be refactored to `options={delegateStatusOptions}` with a `const delegateStatusOptions = ...` definition at the top-level (ie outside the `RoleForm` function) for clarity.

Reply:
Done.